Lib/hmac.py

```diff
@@ -119,10 +124,15 @@ def copy(self):
         """
         # Call __new__ directly to avoid the expensive __init__.
         other = self.__class__.__new__(self.__class__)
-        other._digest_cons = self._digest_cons
         other.digest_size = self.digest_size
-        other._inner = self._inner.copy()
-        other._outer = self._outer.copy()
+        other._name = self._name
+        if self._hmac is not None:
```

Comment:
`if self._hmac:` should be sufficient, truthiness is plenty rather than a full `is not None`.  same elsewhere.